App.tsx

```diff
@@ -1,17 +1,19 @@
+import { initializeLibraries } from "./init";
+
 import { useState } from "react";
 import { Spinner, TamaguiProvider, YStack } from "tamagui";
 
 import { Header } from "./components/Header";
 import { TodoList } from "./components/TodoList";
-import { initializeLibraries } from "./init";
 import { TodoContext, TodoStore } from "./stores/Todo.store";
-import { NewTodoButton } from "./components/NewButton";
 import { observer } from "mobx-react-lite";
+import { NewDialog } from "./components/NewDialog";
+import { TodoStorageGateway } from "./gateways/Todo.gateway";
 
 const { tamaguiConfig } = initializeLibraries();
 
 const ThemedApp = observer(function ThemedApp() {
-  const [store] = useState(() => new TodoStore());
+  const [store] = useState(() => new TodoStore(new TodoStorageGateway()));
```

Comment:
it would be cleaner if you had a function in 'Todo.store' like `getToDoStore()` which sets up the todo store with all its dependencies.

It will keep this cleaner.

Reply:
Done.